src/components/Pagination/Pagination.js

```diff
@@ -155,17 +155,16 @@ export default class Pagination extends Component {
       pageSize: currentPageSize,
     } = state;
     const pageSizesChanged = !equals(pageSizes, prevPageSizes);
+    if (pageSizesChanged && !pageSizes.includes(pageSize)) {
+      pageSize = pageSizes[0];
+    }
     const pageChanged = page !== prevPage;
     const pageSizeChanged = pageSize !== prevPageSize;
     return !pageSizesChanged && !pageChanged && !pageSizeChanged
       ? null
       : {
-          page: pageSizesChanged ? 1 : pageChanged ? page : currentPage,
-          pageSize: pageSizesChanged
-            ? pageSizes[0]
-            : pageSizeChanged
-            ? pageSize
-            : currentPageSize,
+          page: pageSizeChanged ? 1 : pageChanged ? page : currentPage,
```

Comment:
```suggestion
          page: pageSizeChanged && 1 || pageChanged && page || currentPage,
```
pretty much the same comment as in https://github.com/carbon-design-system/carbon/pull/4239

Reply:
I intended to target master